types/database/index.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export interface NotifyProps {
```

Comment:
shouldn't this be a definition file ?

Reply:
These are small definitions and I was thinking of having a file containing several such re-usable interfaces.  Do you rather prefer the approach of splitting them into files?

Reply:
Not really splitting them, but I was talking about the file itself, I believe if you name this `index.d.ts` you wont need to export/import the types.

Perhaps @shazm can chime in here

Reply:
Yup - just discussed this with @avinashlng1080 -

Broadly anything in a `@types` directory, or in a `.d.ts` file gets picked up by VSCode / TS Compiler - and you can call the defined types / interfaces without needing to import or export them.

The exception is when you import or export anything in the `types` or `.d.ts` file, then anything else that is dependant on that interface/type also requires importing / exporting, and this is usually the scenario when needing to import interfaces from 3rd parties.

The quickest way around it is to do something like:

```
import SomeInterface from "some-libary";

declare global {
  interface MyInterface {
    someInterfaces: SomeInterface[];
  }
}
```

And it's usable via:

```
class MyClass implements MyInterface {
  ...
}
```

If you don't wrap it in a `declare global {}` then you need to import / export;

```
import SomeInterface from "some-libary";

export interface MyInterface {
  someInterfaces: SomeInterface[];
}
```

```
import MyInterface from "types";

class MyClass implements MyInterface {
  ...
}
```

Reply:
I'm wondering if it's also worth doing something like:


```
// index.d.ts 

import User from "@typings/User";
import NotifyProps from "@typings/NotifyProps";

declare global {
  UserInterface: User,
  NotifyPropTypes: NotifyProps,
  ...
}
```

^^ Haven't tested this but could be an elegant solution to have one place with all our interfaces/types we want globally available?

Reply:
Thanks @enahum  and @shazm .

I did a couple of tests and here is what I found: 

1.  After renaming the index.ts to index.d.ts, I still had to import the interfaces in the User.ts file.  However, by removing the export keyword before the interfaces did the trick; making the interfaces available 'globally'.  This the reasoning behind: 

```
Basically, TypeScript has two kind of module types declaration: "local" (normal modules) and ambient (global). The second kind allows to write global modules declaration that are merged with existing modules declaration. What are the differences between this files?

d.ts files are treated as an ambient module declarations only if they don't have any imports. If you provide an import line, it's now treated as a normal module file, not the global one, so augmenting modules definitions doesn't work.

Ref: https://stackoverflow.com/questions/39040108/import-class-in-definition-file-d-ts
```
2.  I also tried what @shazm  proposed and it didn't work out as expected.  The definitions in the types/database folder are actually  **classes** with imports from the WDB modules.  Things just broke on TS land (because of point 1), none of the class definitions were available globally and I guess that also the reasoning behind the approach WDB took with its definitions. ( [Nozbe model defintion](https://github.com/Nozbe/WatermelonDB/blob/master/src/Model/index.d.ts) ) .  But the good thing is that the way we've defined the models on our side is similar to WDB and also 'approved' by TS [official docs](https://www.typescriptlang.org/docs/handbook/modules.html).

Reply:
What about using require instead of import? I believe TS has a special syntax for this? This is not a blocker but a nice to have, maybe we could do it later on.

Just for clarity, are we keeping it as is in this PR?

Reply:
Lol nvm, I already merged it so the answer to my question is YES